Client/Frontend/AuthenticationManager/PasscodeEntryViewController.swift

```diff
@@ -37,33 +39,45 @@ class PasscodeEntryViewController: BasePasscodeViewController {
             make.bottom.left.right.equalTo(self.view)
             make.top.equalTo(view.safeArea.top)
         }
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(willEnterForeground),
```

Comment:
I think better than using a global observer here, let's just let the prompting bleed back through from the app delegate. The app delegate already calls into the authenticator to setup this VC through a facade, and we should attempt to leverage this mechanism again. Otherwise we have one mechanism for presentation (e.g. app delegate), and a different mechanism for updating (e.g. notification center).

Currently, `AppAuthenticator` performs an early return when `isPrompting` is true. Instead of performing an early return we should just call something like `pincodeVC.update()` (I know the `AppAuthenticator` currently does not keep a reference to the pincodeVC, but not hard to add IMO. Thoughts?

Reply:
Very good idea, thanks.
`passcodeVC` is already there, but in a static method, a small refactor should solve it